packages/cli/src/auth/auth.service.ts

```diff
@@ -57,9 +58,17 @@ export class AuthService {
 				}
 			}
 		}
-
-		if (req.user) next();
-		else res.status(401).json({ status: 'error', message: 'Unauthorized' });
+		
+		if (req.body) {
+			if (Object.keys(req.body).length === 0) { //oauth flow sends a lot of {} request bodies before the main request (these normally have req.user but not in our case so adding this logic) 
+				next(); 
+			} else if ("type" in req.body && (req.body.type === 'googleCalendarOAuth2Api' || req.body.type === 'microsoftOutlookOAuth2Api')) {
+				next(); 
```

Comment:
good start! as discussed, let's flag by request url instead of auth type so we don't have to change this when we use more credentials